sound/soc/sof/sof-priv.h

```diff
@@ -286,17 +286,17 @@ struct snd_sof_dsp_ops {
 	/* ALSA HW info flags, will be stored in snd_pcm_runtime.hw.info */
 	u32 hw_info;
 
-	const struct sof_arch_ops *arch_ops;
+	const struct dsp_arch_ops *dsp_arch_ops;
 };
 
 /* DSP architecture specific callbacks for oops and stack dumps */
-struct sof_arch_ops {
+struct dsp_arch_ops {
 	void (*dsp_oops)(struct snd_sof_dev *sdev, void *oops);
 	void (*dsp_stack)(struct snd_sof_dev *sdev, void *oops,
 			  u32 *stack, u32 stack_words);
 };
 
-#define sof_arch_ops(sdev) ((sdev)->pdata->desc->ops->arch_ops)
```

Comment:
turns out until now we used the same name `sof_arch_ops` for a structure and a macro, what could possibly go wrong ;-) I think it's good to separate the names, but I'm a bit uncertain about removing the `sof_` prefix from the structure. It is indeed internal, but still maybe it would be better to preserve it... You'd need to use some creativity to invent a new name for it though :-)

Reply:
Yes, that was also strange. I have dropped the sof prefix as the ops are DSP architecture specific, not related to SOF.
The macro retains the sof prefix to make it more consistent with the rest of the code and also to make sure that we do not have name collision.
with the `sof_dsp_arch_ops()` we reach out for the dsp_arch_ops of the SOF device is the reasoning.